src/js/index.js

```diff
@@ -0,0 +1,114 @@
+import "../css/style.css";
+import {shapes} from "./entities/shapes";
+import {Point} from "./entities/points";
+let mousePosition;
+let offsets;
+let currentShape;
+let canvas;
+let context;
+
+window.onload = () => {
+    offsets = [];
+    currentShape = null;
+    canvas = document.getElementById('canvas');
+    context = canvas.getContext("2d");
+    mousePosition = new Point(0, 0);
+
+    canvas.addEventListener("mousemove", (event) => {
+        mousePosition.x = event.offsetX;
+        mousePosition.y = event.offsetY;
+        if(currentShape) {
+            currentShape.move(offsets, mousePosition);
+            drawAllShapes();
+        }
+    });
+    
+    canvas.addEventListener("mousedown", () => {
+        pickShape();
+        calcOffsets();
+    });
+    
+    canvas.addEventListener("mouseup", () => {
+        if(currentShape) {
+            checkShapesIntersection();
+            currentShape = null;
+            offsets = [];
+            drawAllShapes();
+        }
+    });
+    drawAllShapes();
+}
+
```

Comment:
it's a good practice to create specific helpers for the additional logic

Reply:
for example, you can create a `IntersectionHelper` and move all methods that relate to intersection into this file